src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs

```diff
@@ -153,25 +153,30 @@ internal void Abort()
         {
             if (!_isClosed)
             {
-                lock (this)
+                try
                 {
-                    if (!_isClosed && _tcpClient != null)
+                    lock (this)
                     {
-                        //free CBT buffer
-                        if (_channelBindingToken != null)
+                        if (!_isClosed && _tcpClient != null)
                         {
-                            _channelBindingToken.Close();
+                            // free CBT buffer
+                            _channelBindingToken?.Close();
+
+                            // must destroy manually since sending a QUIT here might not be
+                            // interpreted correctly by the server if it's in the middle of a
+                            // DATA command or some similar situation.  This may send a RST
+                            // but this is ok in this situation.  Do not reuse this connection
+                            _tcpClient.LingerState = new LingerOption(true, 0);
```

Comment:
https://github.com/dotnet/runtime/issues/30732#issuecomment-556963448 indicates that this bug manifests due to `LingerState` throwing here.

This leaves the socket open. Would it make more sense to scope the try/catch around just this line? Alternately, a comment indicating that this is not a bug and that the socket will eventually be cleaned up as part of `Dispose()`/`ReleaseConnection()` would be useful.

Reply:
> Would it make more sense to scope the try/catch around just this line?

My understanding is that we're dealing with a race condition; if that's true, I'd expect this could throw but so could some of the subsequent operations.  @tmds can correct me if that's not the case.

Regardless, this exception is causing a bunch of CI and official build test failures, so I'm going to merge.  If you think it can be improved further, can you open a separate issue for it and sync with @tmds on what to do?